src/main/java/rx/subjects/UnicastSubject.java

```diff
@@ -272,8 +309,7 @@ void replay() {
                 if (s != null) {
                     boolean d = done;
                     boolean empty = q.isEmpty();
-
-                    if (checkTerminated(d, empty, s)) {
+                    if (checkTerminated(d, empty, delayError, s)) {
```

Comment:
You should have a local delayError field for this, otherwise you are re-reading the flag on each emission which can cause unnecessary false sharing.

```java
Queue<Object> q = queue;
boolean delayError = this.delayError;
// ...
```